jgrapht-core/src/main/java/org/jgrapht/alg/cycle/ChordalGraphMinimalVertexSeparatorFinder.java

```diff
@@ -0,0 +1,167 @@
+ * </ul>
+ * <p>
+ * Let $\sigma = (v_1, v_2, \dots, v_n)$ be some perfect elimination order (peo) of the graph $G = (V, E)$.
+ * The induced graph on vertices $(v_1, v_2, \dots, v_i)$ with respect to peo $\sigma$ is denoted as $G_i$.
+ * The predecessors set of vertex $v$ with respect to peo $\sigma$ is denoted as $N(v, \sigma)$.
+ * A set $B$ is called a <i>base set</i> with respect to $\sigma$, is there exist some vertex $v$ with
+ * $t = \sigma(v)$ such that $N(v, \sigma) = B$ and B is not a maximal clique in $G_{t-1}$.
+ * The vertices which satisfy conditions described above are called <i>dependent vertices</i>
+ * with respect to $\sigma$. The cardinality of the set of dependent vertices is called a multiplicity
+ * of the base set $B$. The definitions of a base set and a minimal vertex separator in the context of
+ * chordal graphs are equivalent.
+ * <p>
+ * For more information on the topic see: Kumar, P.Sreenivasa &amp; Madhavan, C.E.Veni. (1998).
+ * <a href="https://www.sciencedirect.com/science/article/pii/S0166218X98001231?via%3Dihub">
+ * Minimal vertex separators of chordal graphs</a>. Discrete Applied Mathematics.
+ * 89. 155-168. 10.1016/S0166-218X(98)00123-1.
+ * <p>
+ * The running time of the algorithm is $\mathcal{O}(\omega(G)(|V| + |E|))$, where $\omega(G)$
+ * is the size of the cardinality of a maximum clique of the graph $G$.
+ *
+ * @param <V> the graph vertex type
+ * @param <E> the graph edge type
+ * @author Timofey Chudakov
+ * @see ChordalityInspector
+ * @since April 2018
+ */
+public class ChordalGraphMinimalVertexSeparatorFinder<V, E> {
+    /**
+     * The graph in which minimal vertex separators to searched in
+     */
+    private final Graph<V, E> graph;
+    /**
+     * {@link ChordalityInspector} for testing chordality of the {@code graph}
+     */
+    private final ChordalityInspector<V, E> chordalityInspector;
+    /**
+     * A mapping of minimal separators to their multiplicities
+     */
+    private Map<Set<V>, Integer> minimalSeparators;
+
+    /**
+     * Creates new {@code ChordalGraphMinimalVertexSeparatorFinder} instance. The {@link ChordalityInspector}
+     * used in this implementation uses the default {@link org.jgrapht.traverse.MaximumCardinalityIterator}
+     * iterator
+     *
+     * @param graph the graph minimal separators to search in
+     */
+    public ChordalGraphMinimalVertexSeparatorFinder(Graph<V, E> graph) {
+        this.graph = Objects.requireNonNull(graph);
+        chordalityInspector = new ChordalityInspector<>(graph, ChordalityInspector.IterationOrder.MCS);
+    }
+
+    /**
+     * Computes and returns a mapping of all minimal vertex separators of the {@code graph}
+     * and returns it. Returns null if the {@code graph} isn't chordal.
+     *
+     * @return computed mapping of all minimal separators to their multiplicities, or null if
+     * the {@code graph} isn't chordal
+     */
+    public Map<Set<V>, Integer> getMinimalSeparators() {
```

Comment:
I would make 2 methods:

1. public Set<Set<V>> getMinimalSeparators
2. public Map<Set<V>, Integer> getMiniminalSeparatorsWithMultiplicities

The reason for this is that minimal separators are well known, but I'm not sure how widely known these 'multiplicities' are. So I would recommend to have the `getMinimalSeparators` return something that is widely known and understood.